lib/CommandCall.js

```diff
@@ -53,32 +44,26 @@ class CommandCall {
    * @returns {string}
    */
   toXML() {
-    if (this.type === 'sh') {
-      if (this.options) {
-        this.xml = iXmlNodeShOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error)
-              + this.command + iXmlNodeShClose();
-      } else {
-        this.xml = iXmlNodeShOpen('', '', '', '', '') + this.command + iXmlNodeShClose();
-      }
-    } else if (this.type === 'cl') {
-      const exec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+    if (!availableCommands.includes(this.type)) {
+      throw new Error(`Invalid command type (${this.type}). Valid types: (${availableCommands.join(', ')})`);
+    }
 
-      if (this.options) {
-        this.xml = iXmlNodeCmdOpen(this.options.exec, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeCmdClose();
-      } else {
-        this.xml = iXmlNodeCmdOpen(exec, '', '', '', '', '') + this.command + iXmlNodeCmdClose();
-      }
-    } else if (this.type === 'qsh') {
-      if (this.options) {
-        this.xml = iXmlNodeQshOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeQshClose();
-      } else {
-        this.xml = iXmlNodeQshOpen('', '', '', '', '') + this.command + iXmlNodeQshClose();
-      }
+    if (this.type === 'cl') {
+      const defaultExec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+      this.type = 'cmd';
+      this.xml = `<${this.type} exec='${this.options.exec || defaultExec}'`;
```

Comment:
```suggestion
      this.xml = `<cmd exec='${this.options.exec || defaultExec}'`;
```
No reason to change `this.type`. That would actually cause problems if they re-use it, since it will go through the else code path and not check for rexx execution.

Reply:
https://github.com/IBM/nodejs-itoolkit/pull/103/files#r381568163